clients/cloud/go/producer.go

```diff
@@ -0,0 +1,127 @@
+// =============================================================================
+//
+// Produce messages to Confluent Cloud
+// Using Confluent Golang Client for Apache Kafka
+//
+// =============================================================================
+
+
+package main
+
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+    "./ccloud_lib"
+    "github.com/confluentinc/confluent-kafka-go/kafka"
+    "os"
+    "fmt"
+    "context"
+    "time"
+    "encoding/json"
+)
+
+type RecordValue ccloud_lib.RecordValue
+
+func main() {
+
+    // Initialization
+    config_file, topic := ccloud_lib.ParseArgs()
+    conf := ccloud_lib.ReadCCloudConfig(*config_file)
+
+    // Create Producer instance
+    p, err := kafka.NewProducer(&kafka.ConfigMap{
+	"bootstrap.servers":       conf["bootstrap.servers"],
+	"sasl.mechanisms":         "PLAIN",
+	"security.protocol":       "SASL_SSL",
+	"sasl.username":           conf["sasl.username"],
+	"sasl.password":           conf["sasl.password"]})
+    if err != nil {
+       fmt.Printf("Failed to create producer: %s", err)
+       os.Exit(1)
+    }
+
+    // Create topic if needed
+    a, err := kafka.NewAdminClientFromProducer(p)
+    if err != nil {
+        fmt.Printf("Failed to create new admin client from producer: %s", err)
+        os.Exit(1)
+    }
+    // Contexts are used to abort or limit the amount of time
+    // the Admin call blocks waiting for a result.
+    ctx, cancel := context.WithCancel(context.Background())
+    defer cancel()
+    // Create topics on cluster.
+    // Set Admin options to wait up to 60s for the operation to finish
+    maxDur, err := time.ParseDuration("60s")
+    if err != nil {
+        fmt.Printf("ParseDuration(60s): %s", err)
+        os.Exit(1)
+    }
+    results, err := a.CreateTopics(
+	ctx,
+	// Multiple topics can be created simultaneously
+	// by providing more TopicSpecification structs here.
+	[]kafka.TopicSpecification{{
+		Topic:             *topic,
+		NumPartitions:     1,
+		ReplicationFactor: 3}},
+	// Admin options
+	kafka.SetAdminOperationTimeout(maxDur))
+    if err != nil {
+	fmt.Printf("Failed to create topic: %v\n", err)
+	os.Exit(1)
+    }
+    for _, result := range results {
+        fmt.Printf("%s\n", result)
```

Comment:
...while this is the per-topic error reporting.
This code should check that the topic creation succeeded, e.g: `if result.Error != nil && result.Error.Code() != kafka.ErrTopicAlreadyExists: .. bail out`

Reply:
Why does it need to check that topic creation succeeded?  Wouldn't the previous check report an error if topic creation failed?

```bash
    if err != nil {
        fmt.Printf("Failed to create topic: %v\n", err)
        os.Exit(1)
    }
```

Or, now maybe your previous GH comment makes sense.  You are saying `err` would report the operation failed.  But if topic creation failed, then it would be reported in `results`?  If that is the case, then FYI all my code is almost copy/paste from https://github.com/confluentinc/confluent-kafka-go/blob/master/examples/admin_create_topic/admin_create_topic.go which means that example is wrong too?

Reply:
That example code "handles" per-topic errors simply by printing the results. A proper app would the per-topic result (and this example should do the same to show case.. of course)